### PR TITLE
improve fault tolerance of dataprotection

### PR DESCRIPTION
**Pull request by pawelvds**
- [ ] You've read the [Contributor Guide](https://github.com/fiskaltrust/.github/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/fiskaltrust/.github/blob/main/CODE_OF_CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

{Summary of the changes}

## Description

{Detail}

Fixes #{issue number}

### CODE REVIEW

File: src/fiskaltrust.Launcher/Commands/Common.cs [hunk 1]
```diff
@@ -229,42 +239,88 @@ public static class CommonHandler
...
             Log.Verbose("Loading Curve.");
             var dataProtector = DataProtectionExtensions.Create(accessToken, useFallback: useFallback).CreateProtector(CashBoxConfigurationExt.DATA_PROTECTION_DATA_PURPOSE);
             var clientEcdhPath = Path.Combine(serviceFolder, $"client-{cashboxId}.ecdh");
 
-            if (File.Exists(clientEcdhPath))
-            {
-                return ECDiffieHellmanExt.Deserialize(dataProtector.Unprotect(await File.ReadAllTextAsync(clientEcdhPath)));
-            }
-            else
+            try
```
Comment (volllly) [inline src/fiskaltrust.Launcher/Commands/Common.cs:292]: We have two nested try catches here.. I think we can remove the out one as the inner one will already catch the problems.

File: src/fiskaltrust.Launcher/Commands/Common.cs [hunk 2]
```diff
@@ -229,42 +239,88 @@ public static class CommonHandler
...
-                    clientEcdh = ECDiffieHellmanExt.Deserialize(await File.ReadAllTextAsync(offlineClientEcdhPath));
+                    var protectedData = await File.ReadAllTextAsync(clientEcdhPath);
                     try
                     {
-                        File.Delete(offlineClientEcdhPath);
+                        return ECDiffieHellmanExt.Deserialize(dataProtector.Unprotect(protectedData));
+                    }
+                    catch (Exception ex)
+                    {
+                        Log.Warning($"Could not decrypt ECDH curve, regenerating a new one. Error: {ex.Message}");
```
Comment (volllly) [inline src/fiskaltrust.Launcher/Commands/Common.cs:303]: There is an overload for the logging methods that have the exception as the first parameter.
```suggestion
                        Log.Warning(ex, $"Could not decrypt ECDH curve, regenerating a new one.");
```

File: src/fiskaltrust.Launcher/Commands/DoctorCommand.cs [hunk 3]
```diff
@@ -90,7 +90,9 @@ public static async Task<int> HandleAsync(CommonOptions commonOptions, CommonPro
                 ftCashBoxConfiguration cashboxConfiguration = new();
 
                 if (clientEcdh is null)
-                { }
+                {
+                    Log.Warning("Failed to load ECDH curve. Unable to proceed with Doctor checks.");
```
Comment (volllly) [inline src/fiskaltrust.Launcher/Commands/DoctorCommand.cs:94]:
```suggestion
                    Log.Warning("Failed to load ECDH curve. Skipping some related doctor checks.");
```

**Review by volllly — DISMISSED** [review]

**Comment by pawelvds** [comment]
@volllly finally I resolved conflicts there and improved parts of code ^^, bit too much commits but the ende effet is most important

**Review by volllly — APPROVED** [review]

**Pull request merged**